packages/java/endpoint/src/test/resources/dev/hilla/generator/endpoints/nonnullable/expected-NonNullableEndpoint.ts

```diff
@@ -56,6 +56,10 @@ function _getNotNullReturnType(__init?: EndpointRequestInit): Promise<ReturnType
   return client.call('NonNullableEndpoint', 'getNotNullReturnType', {}, __init);
 }
 
+function _returnNonnullMappedType(__init?: EndpointRequestInit): Promise<Array<string | undefined>> {
```

Comment:
It seems that the `@Nonnull` is still ignored for the type argument. Or am I missing something?

The endpoint returns `@Nonnull Page<@Nonnull String>`, so I’d expect the `Array<string>`, but not `Array<string | undefined>` in the return type for the generated method.

```suggestion
function _returnNonnullMappedType(__init?: EndpointRequestInit): Promise<Array<string>> {
```

Reply:
You're right and unfortunately this is a different issue: while the one I fixed was a regression in the method logic when adding support for `@NonNullApi`, here I see that the code is completely unaware of that annotation as the type has been mapped. I'm searching for a fix.